src/qml/filters/gpsgraphic/ui.qml

```diff
@@ -0,0 +1,1448 @@
+        }
+
+        Label {
+            Layout.topMargin: 10
+            text: qsTr('Preset')
+            Layout.alignment: Qt.AlignRight
+        }
+        Shotcut.Preset {
+            Layout.topMargin: 10
+            id: preset
+            parameters: allParameters
+            onBeforePresetLoaded: {
+                filter.resetProperty("legend_unit")
+            }
+
+            onPresetSelected: {
+                filterRect = filter.getRect(rectProperty)
+                setControls()
+                reset_legend_unit()
+            }
+        }
+
+        Label {
+            topPadding: 10
+            text: qsTr('<b>GPS options</b>')
+            Layout.columnSpan: 2
+        }
+
+        Label {
+            id: gps_sync_major
+            text: qsTr('GPS offset')
+            Layout.alignment: Qt.AlignRight
+            Shotcut.HoverTip { text: qsTr('This is added to video time to sync with gps time.') }
+        }
+        RowLayout {
+            Shotcut.ComboBox {
+                id: combo_majoroffset_sign
+                implicitWidth: 39
+                model: ListModel {
+                    id: sign_val
+                    ListElement { text: '+'; value: 1}
+                    ListElement { text: '-'; value: -1}
+                }
+                currentIndex: 0
+                textRole: 'text'
+                Shotcut.HoverTip { text: qsTr('+ : Adds time to video (use if GPS is ahead).\n - : Subtracts time from video (use if video is ahead).') }
+                onActivated: {
+                    if (_disableUpdate) return
+                    filter.set('majoroffset_sign', sign_val.get(currentIndex).value)
+                    recompute_time_offset()
+                }
+            }
+            Label {
+                text: qsTr('days:')
+                Layout.alignment: Qt.AlignRight
+            }
+            TextField {
+                id: offset_days
+                text: '0'
+                horizontalAlignment: TextInput.AlignRight
+                validator: IntValidator {bottom: 0; top: 36600;}
+                implicitWidth: 25
+                MouseArea {
+                    anchors.fill: parent
+                    onWheel: wheel_offset( wheel.angleDelta.y>0 ? 86400 : -86400 )
+                    onClicked: offset_days.forceActiveFocus()
+                }
+                onFocusChanged: if (focus) selectAll()
+                onEditingFinished: recompute_time_offset()
+                Shotcut.HoverTip { text: qsTr('Number of days to add/subtract to video time to sync them.') }
+            }
+            Label {
+                text: qsTr(' h:')
+                Layout.alignment: Qt.AlignRight
+            }
+            TextField {
+                id: offset_hours
+                text: '0'
+                horizontalAlignment: TextInput.AlignRight
+                validator: IntValidator {bottom: 0; top: 59;}
+                implicitWidth: 20
```

Comment:
I think it's a scaling thing, on my screen I can fit easily 2 digits (this was my intention):
![image](https://user-images.githubusercontent.com/3215812/181593022-7c5c27a9-59d2-46e2-aef6-d7ecfd567f25.png)

I just use the mouse wheel to edit them. 
I don't know what other design choice to make for better usage, the clasic ddd:hh:mm:ss would be a step down in my opinion.
It's already pretty long, but should I increase the width by 50% for each one?

Reply:
I think removing the text allows me to increase width by 50% to 30px and still look decent:
![image](https://user-images.githubusercontent.com/3215812/181638000-bdbd7fbd-b8c1-40b6-b0b8-e3f2b824cd3e.png)

Reply:
I think you could use fontMetrics.advanceWidth("XX") + 5 to have a solution that responds to font size changes.

https://doc.qt.io/qt-6/qml-qtquick-fontmetrics.html#advanceWidth-method

Reply:
I''m not sure why but fontmetrics returns less than I expected, 6px per char. This image is how it looks for me with 2x what it returns. I'd rather go with a fixed 30, what do you think? 2.5x could also get 30 but I don't know what it returns on other PCs, might look ridiculous.

![image](https://user-images.githubusercontent.com/3215812/182965128-d146dca0-e1ec-40be-818f-19a167bd5e6f.png)

Maybe I'm using it wrong? I added a :

```
    FontMetrics {
        id: fontMetrics
        font.family: "Arial"
    }
```

Into the root UI element then used it directly as:
`implicitWidth: fontMetrics.advanceWidth("99")*2`

Reply:
Maybe you are not specifying the correct font. Consider trying:

```
FontMetrics {
    id: fontMetrics
    font: offset_days.font
}
```

I find that I usually need to add a few pixels for padding. Maybe try:

`implicitWidth: fontMetrics.advanceWidth("99") * application.devicePixelRatio + 6`

Play with the "6" to find a good padding.

Reply:
It still returns 6 per char for me. I vote we go with a fixed 30.

Reply:
The 30 works OK on my screen for two digits. But is not enough for triple digit days. Unless you want to add months/years, you might want to support 3 or 4 digits for days.

Also, you should add validators for the fields. Hours should be between 0-23. Minutes between 0-59. Seconds 0-59

![image](https://user-images.githubusercontent.com/821968/183231188-b2584c41-a1c5-4274-b983-c8b26cf99086.png)

Reply:
I do have validators for all the fields but for some reason they don't revert back invalid values. I can see the invalid value doesn't go through (mouse up/down reverts to the previously valid value). I wonder if there's some text(string) vs int interaction I'm not aware of.

`                validator: IntValidator {bottom: 0; top: 59;}`

For the days I'm not sure what to do, I expect this field to be 0 for most "real" use cases but I can see 2 sort of common exceptions: 
-date of the device gets reset to something "default" like 1st of january <manufactured year> which would add some hundreds of years, or even 1xxx
-video (or image) has no date so it's read as epoch 0 which adds... 19.000+ days

I would personally consider both of these cases exceptions and not fully support them (visually I mean) because 5 digits wastes a lot of space for a corner case. I say a decent middle ground is 40px for days to make it slightly different from the others and induce the idea of "this field is different". I really don't want to add months/years, functionally they would serve no real purpose.

Reply:
> I wonder if there's some text(string) vs int interaction I'm not aware of.

I found this helpful idea: https://stackoverflow.com/a/59113560/4355458

This worked for me:

```
                onTextChanged:
                {
                    if(!acceptableInput)
                        offset_hours.undo()
                }
```
But your hours validator is actually wrong. The top should be 23, not 59.

> I would personally consider both of these cases exceptions and not fully support them 

I think you are underestimating our users creativity. Users will add the filters to tracks, color/transparent clips, pictures that they have downloaded off the internet, etc. Putting the filter on another track is particularly helpful because it allow them to apply other affects to the text or graph without applying the effect to the video. They are going to expect all these scenarios to work. Let's just do 4 digits for the days. That should get us pretty far and hopefully avoid bug reports.

I would also encourage you to test some other use cases that are not the specific use case that you have in mind. I notice that when I apply the filter to a track, the filter starts from the beginning with each clip that it encounters on the track. This will be unexpected by users. But I do not know if we have to fix that as part of this PR. But, minimally, the filter should work well on a transparent clip on another track.

Another idea I have to help with the time calculation (not required for this PR):
We could add a status section that shows the detected start time of the clip and the gps file. That would allow the user to see what the expected offset is between the items. Something to think about.

Reply:
> I found this helpful idea: https://stackoverflow.com/a/59113560/4355458
> But your hours validator is actually wrong. The top should be 23, not 59.

Good catch & find.

>Let's just do 4 digits for the days

Ok, works for me.

> Putting the filter on another track is particularly helpful because it allow them to apply other affects to the text or graph without applying the effect to the video. 

This will require tweaks from their part as it's not how the filter can work: it needs to "sync" to something that will provide a time back to it (which is normally the "current" clip underneath) so this is not directly supported simply because of how the entire filter works. I did encounter a case myself when I wanted to do this and I had to visually re-sync the gps (as it was now starting at epoch 0). I don't see another way around this except maybe making a dummy copy of the real clip and making the clip copy transparent somehow.

>I notice that when I apply the filter to a track, the filter starts from the beginning with each clip that it encounters on the track. 

This is actually intentional (unless I misunderstood your case) and I think the correct way to it. In your case the filter starts from the begining because you're reusing the same image/clip which means it will re-sync to that previous "location". 
But think of this scenario: you do a travel vlog in Paris and you add a map showing the current location. Every time you show this one clip near the eiffel tower you definitely want the map to show the same real location near the eiffel tower.

The previous gps_text filter actually messed this up when applied to the timeline by adding the current timeline time to the actual clip time ( mlt_frame_original_position() vs mlt_frame_get_position() ) so the same clip shown at minute 0 and then again at minute 10 would show different values (and I actually thought I had a major bug because locally it was perfectly synced but it somehow drifted when showing another clip, so for a short <1min export this was almost invisible, but for a 30min video it was quite absurd - but only if applied to the timeline, directly on clip was fine).

I'm going to wait for user feedback on variations for this one as I assume people will have specific use cases that I can't even imagine right now and will decide what to support on a case by case/popularity basis.

> We could add a status section that shows the detected start time of the clip and the gps file

Hmm, I actually removed this even from the original gps_text (I left it as a hidden feature when hovering over the chosen .gpx file (see attached img)), I found out in practice that I never ever used it and even when I had a sync issue it was more confusing than just trying to find a "stop" in the video and sync it to a 0km/h speed in gps. It's actually more confusing than it looks because of timezones and the fact that most clocks drift if not connected to the internet.
![image](https://user-images.githubusercontent.com/3215812/183253075-ecb0fa46-645e-4180-a415-92eb6fc84980.png)
vs
![image](https://user-images.githubusercontent.com/3215812/183253210-91e94341-b937-45a3-b545-1a402d83782d.png)

If you think it's useful enough I can add it back from diff. I kind of think the even longer UI might scare people and it looks better without.

Reply:
> I'm going to wait for user feedback on variations for this one as I assume people will have specific use cases that I can't even imagine right now and will decide what to support on a case by case/popularity basis.

I think that is a good approach

> If you think it's useful enough I can add it back from diff. I kind of think the even longer UI might scare people and it looks better without.

I think it is probably useful  because it will help the user to understand why their offset is 150 days or whatever might look strange to them. But let's tackle that in another PR.

Thanks for your continued discussion and consideration in this PR. Let's fix the validation, clean up the nitpicks and get this one merged. I hope you will continue to make improvements in future PRs.

Reply:
> I think it is probably useful because it will help the user to understand why their offset is 150 days or whatever might look strange to them. But let's tackle that in another PR.

All the data is already there so this is just a matter of restoring the old lines. I've added them back quickly.

>Thanks for your continued discussion and consideration in this PR. Let's fix the validation, clean up the nitpicks and get this one merged. I hope you will continue to make improvements in future PRs.

No problem, and sure, I'll hopefully have at least one update with the planned stuff and maybe other user suggestions.